clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/manifest/KubernetesKind.java

```diff
@@ -141,10 +146,24 @@ public static KubernetesKind fromString(String name, boolean registered, boolean
       throw new IllegalArgumentException("The 'NONE' kind cannot be read.");
     }
 
+    final String apiGroup = apiVersion != null ? apiVersion.getApiGroup() : "";
+    final String scopedName = !apiGroup.isEmpty() ? name + "." + apiGroup : "";
+
+
     synchronized (values) {
       Optional<KubernetesKind> kindOptional = values.stream()
-          .filter(v -> v.name.equalsIgnoreCase(name) || (v.alias != null && v.alias.equalsIgnoreCase(name)))
-          .findAny();
+        .filter(v -> v.name.equalsIgnoreCase(scopedName) || v.name.equalsIgnoreCase(name) || (v.alias != null && v.alias.equalsIgnoreCase(name)))
+        // if there are matches on both scopedName and name, prefer scoped name
```

Comment:
One more question: what happens in your case where you have two different kinds with name "networkPolicy"? Does the non-scoped one disappear?